### ISSUE TITLE
`System.Reflection.Metadata` dependency on v4.0

### ISSUE THREAD

**Issue by vaind** [getsentry/sentry-dotnet#2825]
[Originally reported on Discord](https://discord.com/channels/621778831602221064/1172378051074609242)

> With uwp it seems like it requires another extra library that wasn't installed by default

```
System.IO.FileNotFoundException: Could not load file or assembly 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at System.Diagnostics.Internal.PortablePdbReader..ctor()
   at System.Diagnostics.EnhancedStackTrace.GetFrames(StackTrace stackTrace)
   at Sentry.Internal.DebugStackTrace.CreateFrames(StackTrace stackTrace, Boolean isCurrentStackTrace)+MoveNext()
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ReverseIterator`1.MoveNext()
   at Sentry.Internal.DebugStackTrace.Create(SentryOptions options, StackTrace stackTrace, Boolean isCurrentStackTrace)
   at Sentry.Extensibility.SentryStackTraceFactory.Create(Exception exception)
   at Sentry.Internal.MainExceptionProcessor.BuildSentryException(Exception exception, Int32 id, Nullable`1 parentId, String source)
   at Sentry.Internal.MainExceptionProcessor.WalkExceptions(Exception exception, Counter counter, Nullable`1 parentId, String source)+MoveNext()
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ReverseIterator`1.MoveNext()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Sentry.Internal.MainExceptionProcessor.CreateSentryExceptions(Exception exception)
   at Sentry.Internal.MainExceptionProcessor.Process(Exception exception, SentryEvent sentryEvent)
```

**Comment by lucas-zimerman** [comment]
as a note, it only happens on .NET Standard 2.0 but not 2.1 (haven't tested the other .NET frameworks)

**Comment by vaind** [comment]
> as a note, it only happens on .NET Standard 2.0 but not 2.1 (haven't tested the other .NET frameworks)

Any chance you have a reproducible example?

**Comment by lucas-zimerman** [comment]
oh on the sample app, I actually got this
<img width="724" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/8229322/a627776a-80cf-4f5f-9c23-eff2603a4d31">

**Comment by lucas-zimerman** [comment]
https://github.com/lucas-zimerman/SRM_Repro
Build SRM_Repro and you'll notice that the build fails due to mising nuget from Sentry

**Comment by vaind** [comment]
Thanks @lucas-zimerman for the thorough investigation. A fix is pending review in #2843
